pkg/query-service/app/logs/v4/resource_query_builder.go

```diff
@@ -0,0 +1,204 @@
+package v4
+
+import (
+	"fmt"
+	"strings"
+
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+	"go.signoz.io/signoz/pkg/query-service/utils"
+)
+
+// buildResourceFilter builds a clickhouse filter string for resource labels
+func buildResourceFilter(logsOp string, key string, op v3.FilterOperator, value interface{}) string {
+	// we are using lower(labels) as we want case insensitive filtering
+	searchKey := fmt.Sprintf("simpleJSONExtractString(lower(labels), '%s')", key)
```

Comment:
All resource attributes are stored as strings in the main table, so if there is a resource named

`service.name = []{x,y}` then it will be stored as a string, but I got your point, let me add an example test case if possible.